### DISCUSSION TITLE
Workflow: JPG - scantailor TIFF - OCR - PDF

### DISCUSSION THREAD

**Discussion by alfish2000** [ScanTailor-Advanced/scantailor-advanced#33] [Q&A]
Hi!

Any advice for an easy to use (open source) workflow on Windows like that:

Take some images as JPEG (e.g. some documents as photos from smartphone)

Then use scantailor to make small sized black and white TIFFs (already works well)

Then I would like to make a searchable PDF with OCR layers and multiple pages from these TIFF files.

Does anyone have some advice (and open source tool proposals) on how to establish this workflow on Windows?

Thanks in advance for your suggestions

**Comment by FriedrichFroebel** [comment]
For getting a PDF from the TIFF files, you might consider using either

* [`tiff2pdf`](http://www.libtiff.org/man/tiff2pdf.1.html) from *libtiff* (where I am not aware of any native pre-existing Windows builds), followed by [`mutool merge`](https://mupdf.com/docs/manual-mutool-merge.html) from MuPDF or
* [ImageMagick](https://imagemagick.org/) (which provides native Windows binaries, but might lead to memory issues with lots of pages and at least on Linux requires you to change a configuration file for PDF support).

Given the PDF, [`ocrmypdf`](https://ocrmypdf.readthedocs.io) is a simple all-in-one solution for adding the text-layer, based upon the Tesseract engine.

**Comment by derrikF** [comment]
> Then I would like to make a searchable PDF with OCR layers and multiple pages from these TIFF files.
> 
> Does anyone have some advice (and open source tool proposals) on how to establish this workflow on Windows?

Take a closer look at ByteScout PDF Multitool - although it can't create pdf, there are a lot of useful things for working with PDF, including OCR

https://bytescout.com/download/download_freeware.html

https://github.com/bytescout?q=pdf&type=all&language=&sort=

**Comment by alfish2000** [comment]
Thanks for your suggestions so far.

I've made some first steps:

- Installed WSL (Windows Subsystem for Linux https://learn.microsoft.com/en-us/windows/wsl/install)
- Installed OCRmyPDF in WSL (https://ocrmypdf.readthedocs.io/en/latest/installation.html#windows-subsystem-for-linux)
- Installed additional german language pack for OCR (https://ocrmypdf.readthedocs.io/en/latest/languages.html#lang-packs)
- Installed img2pdf in WSL as suggested here: https://ocrmypdf.readthedocs.io/en/latest/cookbook.html#produce-pdf-and-text-file-containing-ocr-text (https://gitlab.mister-muffin.de/josch/img2pdf)

Then I could execute in Windows CMD or PowerShell):
wsl img2pdf *.tif -o out.pdf
wsl ocrmypdf -l deu out.pdf out_ocr.pdf

Input 2 black/white tif files of 2564x3384 size with 45kb and 36kb produced a searchable PDF with 94kb size.


gImageReader (https://github.com/manisandro/gImageReader) also looks promising.
It has an UI and you can check and correct the OCR results before exporting a PDF.

**Comment by zvezdochiot** [comment]
Hi @alfish2000 .

See also [MFBpdf](https://github.com/ImageProcessing-ElectronicPublications/mfbpdf) (Easy converting PNM to (MASK+FG+BG)-pdf).
